docs/logs/correlation/Program.cs

```diff
@@ -32,10 +32,7 @@ public static void Main()
         // Setup Logging
         using var loggerFactory = LoggerFactory.Create(builder =>
         {
-            builder.AddOpenTelemetry(options =>
-            {
-                options.AddConsoleExporter();
-            });
+            builder.AddOpenTelemetry().AddConsoleExporter();
```

Comment:
Regarding the sample in the PR description:
```csharp
builder.Logging
    .ClearProviders()
    .AddEventLog() // Order must be changed
    .AddOpenTelemetry(options => options.ParseStateValues = true)
    .SetResourceBuilder(resourceBuilder)
    .AddConsoleExporter();
```

For the sake of other reviewers, I found it helpful that you pointed out that while it is common for builder methods to be chainable like the [Authorization extensions](https://github.com/dotnet/aspnetcore/blob/f15af7b92d0125f9eba30c8355d82cd6943000e9/src/Security/Authorization/Core/src/AuthorizationServiceCollectionExtensions.cs#L21), even ASP.NET Core has examples where the extensions are not chainable like with the [Authentication extensions](https://github.com/dotnet/aspnetcore/blob/f15af7b92d0125f9eba30c8355d82cd6943000e9/src/Security/Authentication/Core/src/AuthenticationServiceCollectionExtensions.cs#L19).

I think this is good precedent that requiring `AddOpenTelemetry` be at the end is OK.

Reply:
Another good example: [AddMvc](https://learn.microsoft.com/dotnet/api/microsoft.extensions.dependencyinjection.mvcservicecollectionextensions.addmvc). Returns [IMvcBuilder](https://learn.microsoft.com/dotnet/api/microsoft.extensions.dependencyinjection.imvcbuilder).